src/AzureIoTHub.Portal.Infrastructure/ProductionConfigHandler.cs

```diff
@@ -82,6 +82,8 @@ internal ProductionConfigHandler(IConfiguration config)
         public override string IdeasAuthenticationHeader => this.config.GetValue(IdeasAuthenticationHeaderKey, "Ocp-Apim-Subscription-Key");
         public override string IdeasAuthenticationToken => this.config.GetValue(IdeasAuthenticationTokenKey, string.Empty);
 
+        public override string CloudProvider => this.config[CloudProviderKey];
```

Comment:
```suggestion
        public override string CloudProvider => this.config.GetValue(CloudProviderKey, "Azure");
```

Reply:
@kbeaugrand It was decided to not set a default value for the CloudProvider but to generate an exception if the configuration was not present (task #1928). 
We still set Azure as default value ?

Reply:
@delager you're right. We should instead provide the value manually in the e2e-docker-compose.yml file to make the tests working.